plugins/ReactNativeOpenFile/setupBackend.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+'use strict';
+
+import type Bridge from '../../agent/Bridge';
+
+module.exports = function setupRNOpenFile(bridge: Bridge, openFileInEditor: () => void) {
+  bridge.on('rn-open-file', val => {
+    // '__REACT_DEVTOOLS_GLOBAL_HOOK__.$inst'
+    var el = val.split('.').reduce(
+      (prev, curr) => prev ? prev[curr] : null,
+      window
+    );
+    if (!el ||
+      !el._reactInternalInstance ||
+      !el._reactInternalInstance._currentElement
+    ) {
+      return;
+    }
+    var source = el._reactInternalInstance._currentElement._source;
```

Comment:
This code is coupled to React internal representation (which is changing as we speak). Can you rewrite it so that it uses the “data” (see `getData()`) just like the frontend code? This way this won’t break when we switch to Fiber.